data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/export/ExportPartitionWorker.java

```diff
@@ -177,6 +186,9 @@ public void run() {
                             LOG.debug("Write to buffer for line " + (recordCount - DEFAULT_BATCH_SIZE) + " to " + recordCount);
                             recordBufferWriter.writeToBuffer(acknowledgementSet, records);
                             records.clear();
+                            recordBytes.clear();
+                            successItemsCounter.increment(DEFAULT_BATCH_SIZE);
```

Comment:
If there are less DEFAULT_BATCH_SIZE records, the metrics emitted will not be correct. It should min(recordCount, DEFAULT_BATCH_SIZE)

Reply:
I will use `records.size()` since `records` represents the current batch.